src/goto-programs/goto_convert.cpp

```diff
@@ -406,8 +406,8 @@ void goto_convertt::convert_gcc_switch_case_range(
     throw 0;
   }
 
-  const auto lb = numeric_cast<mp_integer>(code.op0());
-  const auto ub = numeric_cast<mp_integer>(code.op1());
+  const auto lb = numeric_cast<mp_integer>(simplify_expr(code.op0(), ns));
+  const auto ub = numeric_cast<mp_integer>(simplify_expr(code.op1(), ns));
 
```

Comment:
Should use make_constant(...)!

Reply:
Possibly so, but `make_constant` lives in the C type checker, and thus isn't available here.